exp/controllers/awsmachinepool_controller.go

```diff
@@ -280,7 +296,7 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		machinePoolScope.Info("starting instance refresh", "number of instances", machinePoolScope.MachinePool.Spec.Replicas)
 		return asgsvc.StartASGInstanceRefresh(machinePoolScope)
 	}
-	if err := reconSvc.ReconcileLaunchTemplate(machinePoolScope, ec2Svc, canUpdateLaunchTemplate, runPostLaunchTemplateUpdateOperation); err != nil {
+	if err := reconSvc.ReconcileLaunchTemplate(machinePoolScope, machinePoolScope, s3Scope, ec2Svc, objectStoreSvc, canUpdateLaunchTemplate, runPostLaunchTemplateUpdateOperation); err != nil {
```

Comment:
why do we need to pass the same `machinePoolScope` twice?

Reply:
It gets cast to two different interfaces (`IgnitionScope` and `LaunchTemplateScope`). The number of small interfaces is to blame here, but many of them are needed to easily mock functionality or share functionality between regular and EKS cluster code and types.